src/libstore/daemon.cc

```diff
@@ -276,8 +276,14 @@ static void performOp(TunnelLogger * logger, ref<Store> store,
     TrustedFlag trusted, RecursiveFlag recursive, unsigned int clientVersion,
     Source & from, BufferedSink & to, unsigned int op)
 {
-    worker_proto::ReadConn rconn { { .from = from } };
-    worker_proto::WriteConn wconn { { .to = to } };
+    worker_proto::ReadConn rconn {
```

Comment:
New commit message:

```
Add protocol versions to {worker,serve}_protocol::*Conn

This will allow us to factor out logic, which is currently scattered inline, into several reusable instances
```